### PR TITLE
Simple generation

### PR DESCRIPTION
**Pull request by davidlgj**
My first stab at a prototype that generates a field from a view shcema and validates it with a validation schema.
It only supports inputs and texareas and lacks a lot at the moment, like not showing error messages when it's pristine, but it's a start. 

It has a starting point directive called forms-js directive that is the startpoint. It kicks of by instantiating a `formsjs.Form` and then builds up child directives depending on `inputType` in the view schema.

``` html
<form forms-js view="view" validation="validation" model="model"></form>
```

The child directives, `fjsText` and `fjsTextarea` then are responsible for building it's part of the form. They both use `ngModel` internally and a `fjsValidate` directive that registers the field with the `formsjs.Form` instance we have.

See `examples/index.html` for more.

### CODE REVIEW

File: source/directives/fjs.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        formsjsForm = new formsjs.Form();
+        this.validationService = formsjsForm.validationService;
+        this.registerAttribute = formsjsForm.registerAttribute.bind(formsjsForm);
+      }],
+      link: function(scope, element, attrs) {
+        scope.$watchGroup(['view','validation'], function(values) {
+          var view  = values[0];
+          var validation = values[1];
+          if (view !== undefined && validation !== undefined) {
+
```
Comment (bvaughn) [inline source/directives/fjs.ts:30]: I assume we're only _requiring_ a view/layout schema in this spike (and it will become optional down the road).

Comment (davidlgj) [inline source/directives/fjs.ts:30]: Yes exactly.

I'd like to support three scenarios:
1. Generate it all <- this is what I've done so far
2. Don't give a view definition but give a validation schema

``` html
<form forms-js validation="validation">
<fieldset>
  <legend>Reqnuired fields</legend>
  <fjs-text key="name"></fjs-text>  
  <fjs-text key="email"></fjs-text>
</fieldset>
<fjs-textarea key="comment"></fjs-textarea>
</form>
```
1. Neither, put validation on each field (Not sure about this.. but hey maybe) I guess we would build a validation schema internally.

``` html
<form forms-js>
<fieldset>
  <legend>Reqnuired fields</legend>
  <fjs-text min="3" required="true" key="name"></fjs-text>  
  <fjs-text pattern=".+@.+\..+" key="email"></fjs-text>
</fieldset>
<fjs-textarea max="270" key="comment"></fjs-textarea>
</form>
```

File: source/directives/validate.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        // We can't hook into the standard validators in angular since we like to
+        // validate after the model has actually gotten updated.
+
+        // TODO: This might need to be a standard watch instead so that it picks up changes
+        //       from the outside scope.
+        scope.ngModel.$viewChangeListeners.push(function() {
+          console.log('Going to validate ', scope.ngModel.$modelValue)
+
+          return attribute.validate()
```
Comment (davidlgj) [inline source/directives/validate.ts:26]: @bvaughn I updated the validate directive to use the `AttributeMetadata.validate()` instead and also bound all fields error messages against `AttributeMetadata.errorMessages` (although I only show the first.)

File: source/directives/fjs.ts (hunk 1 above)
Comment (bvaughn) [inline source/directives/fjs.ts:30]: Interesting. I had not considered use-case 3 but maybe there's value there? Seems like it should be doable if we decide there is.

File: source/directives/validate.ts (hunk 2 above)
Comment (bvaughn) [inline source/directives/validate.ts:26]: Nice!

**Comment by bvaughn** [comment]
Going to merge this because I want to play with it some.

**Pull request merged**